### PR TITLE
chore: update next and platform-nextjs-plugin to latest

### PR DESCRIPTION
**Pull request by BRKalow**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

Updates `next` and `@hashicorp/platform-nextjs-plugin` to their latest versions. This gets us on the most recent next version, and also removes our reliance on `next-transpile-modules` and the long-deprecated `next-optimized-images` plugin.

Upgrading to the latest versions of these packages introduced a bundling issue with `swingset`'s nested `next-mdx-remote` dependency. To get around this, I've updated the root `next-mdx-remote` dependency to effectively be aliased as `next-mdx-remote-v1`. This causes swingset's `next-mdx-remote` version to be hoisted to the root, so Next is able to correctly treat it as an external package.

The above workaround _also_ caused issues with `@hashicorp/react-docs-page`'s `next-mdx-remote` dependency. I've added a local patch to change all usages of `next-mdx-remote` to `next-mdx-remote-v1` in `@hashicorp/react-docs-page`.
## 💭 Anything else?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204725679708958

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -170,9 +172,9 @@
 		"mdast": "^3.0.0",
 		"mdast-util-to-string": "^2.0.0",
 		"moize": "^6.1.0",
-		"next": "^13.0.5",
+		"next": "^13.4.7",
 		"next-auth": "^4.19.2",
-		"next-mdx-remote": "^3.0.8",
+		"next-mdx-remote-v1": "npm:next-mdx-remote@^3.0.8",
```
Comment (thiskevinwang) [inline package.json:177]: COOL

File: patches/@hashicorp+react-docs-page+17.9.0.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  * SPDX-License-Identifier: MPL-2.0
+  */
+ 
+-import type { MDXRemoteSerializeResult } from 'next-mdx-remote'
+-import { serialize } from 'next-mdx-remote/serialize'
++import type { MDXRemoteSerializeResult } from 'next-mdx-remote-v1'
++import { serialize } from 'next-mdx-remote-v1/serialize'
+ import { Pluggable } from 'unified'
+ import grayMatter from 'gray-matter'
+ interface Options {
```
Comment (thiskevinwang) [inline patches/@hashicorp+react-docs-page+17.9.0.patch:28]: sorcery

File: src/components/dev-dot-content/mdx-components/mdx-headings/utils/is-injected-permalink.ts [hunk 3]
```diff
@@ -15,14 +15,13 @@ import { ReactChild, ReactFragment, ReactPortal } from 'react'
  * We expect injected permalink elements to have a `__permalink-h` className.
  * Ref: https://github.com/hashicorp/remark-plugins/blob/d7869b253b3ae07116f280f95fe2b414145594f1/plugins/anchor-links/index.js#L155
  */
-export function isInjectedPermalink(
-	child: ReactChild | ReactFragment | ReactPortal
-): boolean {
+export function isInjectedPermalink(child: ReactNode): boolean {
```
Comment (thiskevinwang) [inline src/components/dev-dot-content/mdx-components/mdx-headings/utils/is-injected-permalink.ts:18]: nice Type update

File: src/components/branded-card/index.tsx [hunk 4]
```diff
@@ -20,7 +21,7 @@ import patternVault from './img/vault.png'
 import patternWaypoint from './img/waypoint.png'
 import s from './branded-card.module.css'
 
-const PATTERN_IMG_MAP: Record<ProductSlug, string> = {
+const PATTERN_IMG_MAP: Record<ProductSlug, StaticImageData> = {
```
Comment (thiskevinwang) [inline src/components/branded-card/index.tsx:24]: 👍

File: package.json [hunk 5]
```diff
@@ -204,7 +206,8 @@
 	},
 	"overrides": {
 		"react": "^18.2.0",
-		"react-dom": "^18.2.0"
+		"react-dom": "^18.2.0",
+		"next-mdx-remote": "^4.2.0"
```
Comment (thiskevinwang) [inline package.json:210]: Is this override for `swingset`'s internal next-mdx-remote?

**Review by thiskevinwang — APPROVED** [review]
This looks very straightforwards to me! All code changes look to be simply updating the `next-mdx-remote` alias.

Any README breadcrumbs for why we do this aliasing (and resulting `patch-package`-ing) would be great

File: package.json (hunk 5 above)
Comment (BRKalow) [reply package.json:210]: This is mainly to satisfy npm, as it warns on each install that `react-docs-page`'s `next-mdx-remote` peer dependency is not met (as it relies on `3.x`)

**Pull request merged**